dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpCdiExtension.java

```diff
@@ -23,20 +28,44 @@
 public class McpCdiExtension implements Extension {
 
     private ToolRegistry tools = new ToolRegistry();
+    private ConcurrentHashMap<String, LinkedList<String>> duplicateToolsMap = new ConcurrentHashMap<>();
 
     void registerTools(@Observes ProcessManagedBean<?> pmb) {
         // TODO: limit this to just bean types with Tool annotations?
         AnnotatedType<?> type = pmb.getAnnotatedBeanClass();
+        Class<?> javaClass = type.getJavaClass();
         for (AnnotatedMethod<?> m : type.getMethods()) {
             Tool toolAnnotation = m.getAnnotation(Tool.class);
             if (toolAnnotation != null) {
-                registerTool(toolAnnotation, pmb.getBean(), m);
+                registerTool(toolAnnotation, pmb.getBean(), m, javaClass.getPackageName() + "." + javaClass.getSimpleName());
```

Comment:
I don't think this will work for inner classes.

```suggestion
                registerTool(toolAnnotation, pmb.getBean(), m, javaClass.getName());
```